components/verse/ToastNotification/ToastNotification.tsx

```diff
@@ -0,0 +1,57 @@
+import { useState, useEffect } from 'react'
+import { BsCheckCircle, BsInfoCircle, BsX, BsXCircle } from 'react-icons/bs'
+
+interface Props {
+   type: 'success' | 'info' | 'error'
+   title: string
+   visibility?: boolean
+   children: string
+}
+
+type BgColors = 'bg-[#ffffff]' | 'bg-[#b6f8c4]' | 'bg-[#b2e6f5]' | 'bg-[#ffb7b7]'
+
+const ToastNotification: React.FC<Props> = ({ type, title, visibility, children }) => {
+   const [showToast, setShowToast] = useState(visibility)
+   useEffect(() => setShowToast(visibility), [visibility])
+
+   const [backgroundHex, setBackgroundHex] = useState<BgColors>('bg-[#ffffff]')
+   useEffect(() => {
+      type === 'success' && setBackgroundHex('bg-[#b6f8c4]')
+      type === 'info' && setBackgroundHex('bg-[#b2e6f5]')
+      type === 'error' && setBackgroundHex('bg-[#ffb7b7]')
+
+      setTimeout(() => setShowToast(false), 5000)
```

Comment:
Possible problem here! This is only running when the component is first rendered, but if the component needs to be used a second time, then it wouldn't disappear because the `setTimeout` will not be called again. I think you should add this inside of a `useEffect` that has `showToast` as dependency so if it's set to true again you have a way to make it disappear once more. Remember that you only want to run this `setTimeout` if the `showToast` is true, otherwise skip it.